climada/engine/supplychain.py

```diff
@@ -0,0 +1,395 @@
+
+        if not user_data:
+            try:
+                # if results folder exists but file does not - > download file
+                if file_name not in os.listdir(results_folder):
+                    os.chdir(SYSTEM_DIR)
+                    download_link = WIOD_FILE_LINK + file_name
+                    download_file(download_link)
+                    LOGGER.debug('Downloading WIOT table for year %s', year)
+
+                    mriot = pd.read_excel(os.path.join(results_folder, file_name),
+                                  engine='pyxlsb')
+
+                # if folder and file exist -> load file
+                else:
+                    mriot = pd.read_excel(os.path.join(results_folder, file_name),
+                                  engine='pyxlsb')
+
+            # if folder does not exist -> create folder and download file
+            except FileNotFoundError:
+                os.chdir(SYSTEM_DIR)
+                download_link = WIOD_FILE_LINK + file_name
+                download_file(download_link)
+                LOGGER.debug('Downloading WIOT table for year %s', year)
+
+                mriot = pd.read_excel(os.path.join(results_folder, file_name),
+                                  engine='pyxlsb')
+        else:
+            mriot = pd.read_excel(os.path.join(file_path, file_name),
+                                  engine='pyxlsb')
+
+        row_st,row_end=rows_range
+        col_sect,row_sect_end=cols_sect_range
+        col_data_st,col_data_end=cols_data_range
+
+        sectors = mriot.iloc[row_st:row_sect_end, col_sect].values
+        countries_iso3 = np.unique(mriot.iloc[row_st:row_end, col_iso3])
+        # move Rest Of World (ROW) at the end of the array as countries
+        # are in chronological order
+        idx_row = np.where(countries_iso3 == 'ROW')[0][0]
+        countries_iso3 = np.hstack([countries_iso3[:idx_row],
+                                    countries_iso3[idx_row+1:], np.array('ROW')])
+
+        mriot_data = mriot.iloc[row_st:row_end,
+                                col_data_st:col_data_end].values
+        total_prod = mriot.iloc[row_st:row_end, -1].values
+
+        n_countries = len(countries_iso3)
+        n_sectors = len(sectors)
+
+        cntry_pos = {}
+        countries = []
+        for i, _ in enumerate(countries_iso3):
+            iso3 = countries_iso3[i]
+            cntry_pos.update({iso3: range(i*n_sectors, n_sectors*(i+1))})
+
+            try:
+                countries.append(countries_by_alpha3[iso3][0])
+            except KeyError:
+                countries.append('Rest of World')
+
+        self.mriot_data = mriot_data
+        self.countries = np.array(countries, dtype=str)
+        self.countries_iso3 = countries_iso3
+        self.sectors = sectors
+        self.total_prod = total_prod
+        self.n_countries = n_countries
+        self.n_sectors = n_sectors
+        self.cntry_pos = cntry_pos
+        self.mriot_type = 'wiod'
```

Comment:
Informational, but currently there are no other choices than using World Input-Output Tables (WIOT) from the [World Input-Output Database (WIOD) project](http://www.wiod.org/home) so it is not that informative. If more tables will be added then we will need another read_table function and the mriot_type could become more informative.

Reply:
Good idea to include this. Please make this clear in the docstring.

Reply:
There is, along with references.

Reply:
What commit? Also, please indicate hwat references you mean. Finally, the docstring  Also, the docstring is not understandable currently (and I think incorrect):
```
    mriot_type : str
        Type of the adopted input-output table. Currently only WIOD tables
        are supported.
```
The code support other tables than 'WIOD'. It is just that only for WIOD tables there is automatic data retrieval. Furthermore, `Type of the adopted input-output table` furthermore is unclear. Does it mean "sources of the data"?

Reply:
That is not correct. The code is hardcoded for WIOD tables only. This is made clear since the very name of the function (read_wiod16 - there is no other function which reads IO tables), as well as repeated in the docstring description. References (bibliography) are even provided in the function description.

Reply:
But I do not understand why `calc_sector_direct_impac` and `calc_sector_indirect_impac` would not work for any other WIOT table ... or rather, these methods should work irrespective of the data source, as long as the data is loaded in the same way as does `read_wiot16`.

Reply:
Those functions will indeed work - but currently the only type of IO we read are WIOD; thus the quote

> Currently only WIOD tables are supported.

even if `calc_sector_direct_impac` and `calc_sector_indirect_impac `funcs are general, you can currently only use them for WIOD tables

Reply:
That is incorrect formulation. The class works very well even without WIOD tables. In this case one however must supply that data for `self.sectors`, `self.countries_iso3`, `self.mriot_data`, `self.total_prod`, `self.cntry_pos`  and `self.mriot_type`.